xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2975,6 +2979,14 @@ void CVideoPlayer::HandleMessages()
       CLog::Log(LOGDEBUG, "CVideoPlayer - CDVDMsg::PLAYER_ABORT");
       m_bAbortRequest = true;
     }
+    else if (pMsg->IsType(CDVDMsg::PLAYER_SET_UPDATE_STREAM_DETAILS))
+    {
+      CDVDMsgPlayerSetUpdateStreamDetails* msg = static_cast<CDVDMsgPlayerSetUpdateStreamDetails*>(pMsg);
```

Comment:
here we need to check if path matches m_fileItem. if not, don't modify m_UpdateStreamDetails.

Reply:
With this check we can tell VP to get stream details only for current playing item, and in this case we don't need std::set and pass file item to new interface method. I wanted to make a universal solution, but if we are going to make this only for current item, then I can do this:
1. Use "bool m_UpdateStreamDetails" instead of std::set
2. CVideoPlayer::SetUpdateStreamDetails is used only for current item and CFileItem dont' pass to it
3. Moreover I can make SetUpdateStreamDetails only to set this flag (application only sets it)

Is it ok?

Reply:
I don't get 3. Please elaborate.

Reply:
Do we need to set and unset this flag (should VP update stream details)? Application only sets it.
VP method should be
``` 
CVideoPlayer::SetUpdateStreamDetails(bool update);
// update=false - clear flag (do not update)
// update=true - set flag (do update)
```
or
```
CVideoPlayer::SetUpdateStreamDetails();
// just set this flag (do update)
```

Reply:
just setting the flag is fine as long unset is not required.